src/Controller/ModelCollector.php

```diff
@@ -241,6 +250,49 @@ public function searchParentOf(ModelInterface $model)
         throw new DcGeneralInvalidArgumentException('Invalid condition, not in hierarchical mode!');
     }
 
+    /**
+     * Search the parent model from a hierarchical model.
+     *
+     * @param ModelInterface $model The hierarchical model for search the parent model.
+     *
+     * @return ModelInterface|null
+     *
+     * @throws DcGeneralInvalidArgumentException It throws a exception if the configuration not passed.
+     */
+    public function searchParentFromHierarchical(ModelInterface $model): ?ModelInterface
+    {
+        if (null === $this->rootProvider) {
+            throw new DcGeneralInvalidArgumentException(
+                'Invalid configuration. The root data provider must be defined!'
+            );
+        }
+        if (null === $this->parentProvider) {
+            throw new DcGeneralInvalidArgumentException(
+                'Invalid configuration. The parent data provider must be defined!'
+            );
+        }
+        if ($this->rootProviderName !== $model->getProviderName()) {
+            throw new DcGeneralInvalidArgumentException(
+                'Model originates from ' . $model->getProviderName() .
+                ' but is expected to be from ' . $this->rootProviderName .
+                ' can not determine parent.'
+            );
+        }
+
+        $condition = $this->relationships->getChildCondition($this->parentProviderName, $this->rootProviderName);
+        if ([] !== ($inverseFilter = $condition->getInverseFilterFor($model))) {
```

Comment:
Shouldn't this rather be:
```diff
- if ([] !== ($inverseFilter = $condition->getInverseFilterFor($model))) {
+ if (null !== ($inverseFilter = $condition->getInverseFilterFor($model))) {
```
As I read #541

Reply:
See here https://github.com/contao-community-alliance/dc-general/pull/541/commits/e50422326a7aba8a9eee302b22cff60f60540f96